### DISCUSSION TITLE
PV standard deviation

### DISCUSSION THREAD

**Discussion by masterfelu** [caproto/caproto#841] [Q&A] [answered]
Hello.
I was wondering if it possible to add the `standard deviation` parameter along with `value` parameter for a PV, as shown on [Phoebus documentation](https://control-system-studio.readthedocs.io/en/latest/app/databrowser/doc/index.html)
<img width="834" alt="Screenshot 2024-03-17 at 4 45 53 PM" src="https://github.com/caproto/caproto/assets/16156847/65ee2250-72b2-49b1-bd58-fdc9088fd2e9">

**Comment by tacaswell** [comment, marked as answer]
That image is looking at data pulled from an archiver, so it has a time series, rather than from an IOC PV.  

In caproto if you want to add additional fields you can, but you would have to implement the statistics your self.

**Reply by masterfelu** [reply]
thanks. i was thinking standard deviation is an additional field readily available for PVs, but it might not be so.

**Reply by tacaswell** [reply]
I think there are too many degrees of freedom to be a good standard thing: over what time window? fixed lag time, manually reset, ...;  do you weight by sample (e.g. each of N values has equal weight) or by time (the value has the weight of how long it was until it changed?) [although you might have to do by time as epics by default suppresses camonitor updates when the value hasn't changed] or by a combination of one of those + age?

I could see there being a couple of cute mix-ins / helper classes to implement each of those schemes.....
